packages/enzyme-test-suite/test/ReactWrapper-spec.jsx

```diff
@@ -647,6 +651,309 @@ describeWithDOM('mount', () => {
+      });
+    });
+
+    describe('useEffect with cleanup Effect', () => {
+      const ChatAPI = {
+        subscribeToFriendStatus: sinon.stub(),
+        unsubscribeFromFriendStatus: sinon.stub(),
+      };
+      function FriendStatus({ friend = {}}) {
+        const [isOnline, setIsOnline] = useState(null);
+      
+        function handleStatusChange(status) {
+          setIsOnline(status.isOnline);
+        }
+      
+        useEffect(() => {
+          ChatAPI.subscribeToFriendStatus(friend.id, handleStatusChange);
+          return function cleanup() {
+            ChatAPI.unsubscribeFromFriendStatus(friend.id, handleStatusChange);
+          };
+        });
+      
+        if (isOnline === null) {
+          return 'Loading...';
+        }
+        return isOnline ? 'Online' : 'Offline';
+      };
+
+      let wrapper;
+      const friend = { id: 'enzyme'}
+      let simulateChange;
+      before(() => {
+
+        wrapper = mount(<FriendStatus friend={friend}/>);
+
+      });
+
+      it('on initial mount', () => {
+        expect(wrapper.debug()).to.eql(
+          `<FriendStatus friend={{...}}>
+  Loading...
+</FriendStatus>`);
+        // TODO: uncomment later after fix
+        // expect(wrapper.html()).to.eql(`Loading...`);
+        expect(ChatAPI.subscribeToFriendStatus.calledOnceWith(friend.id)).to.true;
+        simulateChange = ChatAPI.subscribeToFriendStatus.args[1];
+      });
+
+      it('simulate status Change', () => {
+        simulateChange({ isOnline: true });
+        expect(wrapper.html()).to.eql(`Online`);
+      })
+      it('simulate status Change again', () => {
+        simulateChange({ isOnline: false });
+        expect(wrapper.html()).to.eql(`Offline`);
+      })
+      it('cleanup on unmount', () => {
+        wrapper.unmount();
+        expect(ChatAPI.unsubscribeFromFriendStatus.calledOnceWith(friend.id)).to.true;
+      });
+    });
+
+    describe('custom hook : useCounter', () => {
+      function useCounter({initialCount = 0, step = 1} = {}) {
+        const [count, setCount] = useState(initialCount)
+        const increment = () => setCount(c => c + step)
+        const decrement = () => setCount(c => c - step)
+        return {count, increment, decrement}
+      }
+      // testing custom hooks with renderProps
+      // may be we can think of adding in utils
+      // will be repeatedl
+      const Counter = ({children, ...rest}) => children(useCounter(rest));
+  
+      function setup(props) {
+        const returnVal = {}
+        render(
+          <Counter {...props}>
+            {val => {
+              Object.assign(returnVal, val)
+              return null
+            }}
+          </Counter>,
+        )
+        return returnVal
+      }
+
+      it('useCounter', () => {
+        const counterData = setup();
+        counterData.increment();
+        expect(counterData.count).to.eq(1);
+        counterData.decrement();
+        expect(counterData.count).to.eq(0);
```

Comment:
for all assertions of the form `expect(obj.property).to.equal(value)`, please use `expect(obj).to.have.property('property', value)`.

Reply:
make sense. Done